src/sentry/static/sentry/app/components/versionHoverCard.jsx

```diff
@@ -17,99 +15,78 @@ import RepoLabel from 'app/components/repoLabel';
 import TimeSince from 'app/components/timeSince';
 import withApi from 'app/utils/withApi';
 
-const VersionHoverCard = createReactClass({
-  displayName: 'VersionHoverCard',
-
-  propTypes: {
+class VersionHoverCard extends React.Component {
+  static propTypes = {
     api: PropTypes.object,
     version: PropTypes.string.isRequired,
     orgId: PropTypes.string.isRequired,
     projectId: PropTypes.string.isRequired,
-  },
+  };
 
-  getInitialState() {
-    return {
+  constructor(props) {
+    super(props);
+    this.state = {
       loading: true,
       error: false,
       data: {},
       visible: false,
       hasRepos: false,
       deploys: [],
+      release: null,
     };
-  },
+  }
 
   componentDidMount() {
     this.fetchData();
-  },
+  }
 
   fetchData() {
-    const {orgId, projectId, version} = this.props;
-    const done = _.after(3, () => {
-      this.setState({loading: false});
-    });
+    const {api, orgId, projectId, version} = this.props;
 
     // releases
     const releasePath = `/projects/${orgId}/${projectId}/releases/${encodeURIComponent(
       version
     )}/`;
-    this.props.api.request(releasePath, {
-      method: 'GET',
-      success: data => {
-        this.setState({
-          release: data,
-        });
-      },
-      error: () => {
-        this.setState({
-          error: true,
-        });
-      },
-      complete: done,
-    });
+    const releaseRequest = api
+      .requestPromise(releasePath, {
+        method: 'GET',
+      })
+      .then(data => {
+        this.setState({release: data});
+      });
 
     // repos
-    const repoPath = `/organizations/${orgId}/repos/`;
-    this.props.api.request(repoPath, {
-      method: 'GET',
-      success: data => {
-        this.setState({
-          hasRepos: data.length > 0,
-        });
-      },
-      error: () => {
-        this.setState({
-          error: true,
-        });
-      },
-      complete: done,
-    });
+    const repoRequest = api
+      .requestPromise(`/organizations/${orgId}/repos/`, {
+        method: 'GET',
+      })
+      .then(data => {
+        this.setState({hasRepos: data.length > 0});
+      });
 
     //deploys
     const deployPath = `/organizations/${orgId}/releases/${encodeURIComponent(
       version
     )}/deploys/`;
-    this.props.api.request(deployPath, {
-      method: 'GET',
-      success: data => {
-        this.setState({
-          deploys: data,
-        });
-      },
-      error: () => {
-        this.setState({
-          error: true,
-        });
-      },
-      complete: done,
-    });
-  },
+    const deployRequest = api
+      .requestPromise(deployPath, {
+        method: 'GET',
+      })
+      .then(data => {
+        this.setState({deploys: data});
+      });
+
+    Promise.all([releaseRequest, repoRequest, deployRequest])
+      .then(() => this.setState({loading: false}))
```

Comment:
Since loading blocks the UI from rendering anyway, we could set all the data here (and use `async/await`), but this is fine too.

Reply:
If I use async/await won't each request happen serially? I wanted to preserve all requests happening in parallel.

Reply:
It depends where you await. You can await the promise.all